src/eventstream-proxy/eventstream-proxy.base.ts

```diff
@@ -136,17 +137,25 @@ export abstract class EventStreamProxyBase extends WebSocketEventsBase {
       }
 
       for (const listener of this.eventListeners) {
-        try {
-          await listener.onEvent(subName, event, (msg: WebSocketMessage | undefined) => {
-            if (msg !== undefined) {
-              messages.push(msg);
-            }
-          });
-        } catch (err) {
-          this.logger.error(`Error processing event: ${err}`);
-        }
+        // Some events require enrichment that could involve a call to the blockchain,
+        // so we don't want to do those synchronously. Create a promise for each onEvent()
+        // handler and when they're all complete we'll create the batch message
+        eventHandlers.push(Promise.resolve(listener.onEvent(subName, event)));
       }
     }
+
+    // Now we need to await the promises in order so the messages stay in order
+    for (const nextProm of eventHandlers) {
+      await nextProm
```

Comment:
You could use a normal `try-catch` around this, rather than using thenables.

Reply:
Agreed - updated in the new commit